prow/config/jobs.go

```diff
@@ -264,6 +262,46 @@ func (cm RegexpChangeMatcher) RunsAgainstChanges(changes []string) bool {
 	return false
 }
 
+// ShouldRun determines if the postsubmit should run against a specific
+// base ref, or in response to a set of changes. The latter mechanism
+// is evaluated lazily, if necessary.
+func (ps Postsubmit) ShouldRun(baseRef string, changes ChangedFilesProvider) (bool, error) {
+	if !ps.RunsAgainstBranch(baseRef) {
+		return false, nil
+	}
+	if ps.RunIfChanged != "" {
+		changeList, err := changes()
+		if err != nil {
+			return false, err
+		}
+		return ps.RunsAgainstChanges(changeList), nil
+	}
+	return true, nil
+}
+
+// ShouldRun determines if the presubmit should run against a specific
+// base ref, in response to a comment body or a set of changes. The latter
+// two mechanisms are evaluated lazily, if necessary.
+func (ps Presubmit) ShouldRun(baseRef string, body string, changes ChangedFilesProvider) (bool, error) {
```

Comment:
I think this is impossible because of the ordering -- we need to check the branch requirements, _then_ the presubmit-specific bits, _then_ optionally the `run_if_changed`. They are not technically related in a superset/subset relationship